Library/Formula/dmd.rb

```diff
@@ -65,6 +57,27 @@ def install
     end
   end
 
+  # Fix up dmd.conf to point to the correct include paths.
+  def fix_dmd_conf
+    # Homebrew's etc.install method doesn't let you overwrite etc files,
+    # so edit it in place if it's there:
+    conf = etc/"dmd.conf"
+    if conf.exist?
+      inreplace conf, /^DFLAGS=.+$/, "DFLAGS=-I#{include}/d2 -L-L#{lib}"
+    else
+      conf.write <<-EOS.undent
+        [Environment]
+        DFLAGS=-I#{include}/d2 -L-L#{lib}
+        EOS
+    end
+
+  end
+
+  def post_install
+    # We need to fix up dmd.conf even if users install/upgrade via bottle:
+    fix_dmd_conf
```

Comment:
> It shouldn't need to be.

Hmm.  Even with the changes you recommend here (which I plan to implement this evening), I'll still need a "fixup" function that gets called both in `install` and in `post_install`.

That's because `etc.install` won't overwrite previous dmd.conf files, so they could still be pointing to old, incorrect library paths. That would break the build, so we have to fix it up before we continue w/ the build.  And of course we need it in `post_install` too, or else it won't run for installs from bottles.